app/jobs/cache/feed_job.rb

```diff
@@ -11,38 +11,34 @@ class FeedJob < ApplicationJob
     queue_as :cache
 
     def perform(url, download_media = false)
-      @url = url
-
-      @feed = begin
+      feed = begin
         benchmark("[Feedjira] #{url}", level: :info) do
           Feedjira::Feed.fetch_and_parse(url)
         end
       end
 
       feed_cache_key = Feed.find_by_url(url).cache_key
       cached_feed = Rails.cache.fetch(feed_cache_key)
-      if cached_feed.blank? || cached_feed.last_modified != @feed.last_modified
-        Rails.cache.write(feed_cache_key, @feed)
-        @updated = true
+      if cached_feed.blank? || cached_feed.last_modified != feed.last_modified
+        Rails.cache.write(feed_cache_key, feed)
+        updated = true
       end
+      after_perform(feed, url, download_media) if updated
+    end
 
+    # Global nav uses some feeds, and is cached so needs to be expired when those
+    # feeds are updated.
+    # Cached pages need to be expired should they be using the updated feed.
+    def after_perform(feed, url, download_media)
```

Comment:
This looks to be a (my?) mistake in the original writing of the `after_perform` callback.

If you look at http://guides.rubyonrails.org/v4.2/active_job_basics.html#callbacks, it only shows callbacks implemented as executions of class methods, not as definitions of instance methods.

Could we therefore rewrite this to use the (proper) form:
```ruby
module Cache
  class FeedJob < ApplicationJob
    # ...

    after_perform do |job|
      # enqueue download media jobs
      # enqueue cache expiration jobs
    end
  end
end
```